pkg/controllers/clusterinventory/clusterprofile/controller.go

```diff
@@ -0,0 +1,265 @@
+/*
+Copyright (c) Microsoft Corporation.
+Licensed under the MIT license.
+*/
+
+// Package clusterprofile features a controller to generate clusterprofile objects from MemberCluster
+package clusterprofile
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/api/meta"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/client-go/util/workqueue"
+	"k8s.io/klog/v2"
+	clusterinventory "sigs.k8s.io/cluster-inventory-api/apis/v1alpha1"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/controller/controllerutil"
+	"sigs.k8s.io/controller-runtime/pkg/event"
+	"sigs.k8s.io/controller-runtime/pkg/handler"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+
+	clusterv1beta1 "go.goms.io/fleet/apis/cluster/v1beta1"
+	"go.goms.io/fleet/pkg/utils"
+)
+
+const (
+	// clusterProfileCleanupFinalizer is the finalizer added to a MemberCluster object if
+	// a corresponding ClusterProfile object has been created.
+	clusterProfileCleanupFinalizer = "kubernetes-fleet.io/cluster-profile-cleanup"
+
+	// the list of reasons in the cluster profile status
+	clusterNoStatusReason      = "MemberAgentReportedNoStatus"
+	clusterHeartbeatLostReason = "MemberAgentHeartbeatLost"
+	clusterHealthUnknownReason = "MemberAgentReportedNoHealthInfo"
+	clusterUnHealthyReason     = "MemberClusterAPIServerUnhealthy"
+	clusterHealthyReason       = "MemberClusterAPIServerHealthy"
+)
+
+// Reconciler reconciles a MemberCluster object and creates the corresponding ClusterProfile
+// object in the designated namespace.
+type Reconciler struct {
+	client.Client
+	ClusterProfileNamespace   string
+	ClusterUnhealthyThreshold time.Duration
+}
+
+// Reconcile processes the MemberCluster object and creates the corresponding ClusterProfile object
+// in the designated namespace.
+func (r *Reconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Result, error) {
+	mcRef := klog.KRef(req.Namespace, req.Name)
+	startTime := time.Now()
+	klog.V(2).InfoS("Reconciliation starts (cluster profile controller)", "MemberCluster", mcRef)
```

Comment:
```suggestion
	klog.V(2).InfoS("Reconciliation starts (cluster profile controller)", "memberCluster", mcRef)
```
klog key starts with lower case.

Reply:
it seems that key in most of the files have it starts with upper case, the doc I found didn't spell it out. 
https://github.com/kubernetes/community/blob/master/contributors/devel/sig-instrumentation/logging.md

Do you have a link?

Reply:
The example used https://github.com/kubernetes/community/blob/master/contributors/devel/sig-instrumentation/logging.md is using lower case,

eg. klog.InfoS("Received HTTP request", "**method**", "GET", "**URL**", "/metrics", "**latency**", time.Second)

URL is different as it's Initialisms.

It's called out in this doc, https://github.com/kubernetes/community/blob/master/contributors/devel/sig-instrumentation/migration-to-structured-logging.md#name-arguments